### ISSUE TITLE
Updating a plugin "outside" the plugin manager appears to break the plugin state cache

### ISSUE THREAD

**Issue by sylae** [openplanet-nl/plugin-manager#23]
This could be a factor of having two Plugin Managers running (one being the one shipped with openplanet, the other being my dev copy), but just in case this is indicative of a possible issue with the plugin cache i thought i'd report this.

I updated Editor Trails to 1.1 using my "dev" version of the plugin manager, and this appears to have messed up the "normal" manager's state and it remains convinced that v1.0 is installed. This persists through trackmania restarts as well as when just the "normal" manager is installed.

anyway obviously this is a bit of an edge case, but like i said, idk if it's just a symptom of something that Might Actually Be An Issue For Normal People, yknow?

![image](https://user-images.githubusercontent.com/2791628/219960793-66306f6f-974d-4349-83c3-e4be53092d0e.png)

**Comment by skybaks** [comment]
I just ran into this issue. I have a plugin folder, but I briefly tested with a zipped version of the plugin. After I was done testing I deleted the zipped version.
Now, the plugin manager is notifying me of updates even though my plugin folder is up to date with the published version.
This is because updating the PluginCache.json file is skipped for non-zipped plugins so the old version I briefly tested with is stuck in the cache file with no way of getting updated or removed.
As part of a potential solution to this it would be nice if there were more checks about what the plugin source was for any given loaded plugin before notifying of an update.

**Comment by skybaks** [comment]
I fixed my issue by manually deleting the PluginCache.json file when the game was closed. What is the purpose of saving plugin versions to this file? Wouldn't it be easy enough to read all the plugin version into memory once when the game starts up?

**Comment by skybaks** [comment]
I'm still seeing this issue. I would happily open a pull request to delete the cache file but I don't want to do that without understanding why it existed to begin with.

**Comment by codecat** [comment]
> What is the purpose of saving plugin versions to this file? Wouldn't it be easy enough to read all the plugin version into memory once when the game starts up?

No, because plugins won't exist in memory if they are unloaded (for example, they are blocked by the server).
